### PR TITLE
[Snyk] Fix for 1 vulnerabilities

### PR DESCRIPTION
**Pull request by ArduinoBot**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `yarn` dependencies of this project.</h3>

#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - arduino-ide-extension/package.json

<details>
<summary>⚠️ <b>Warning</b></summary>

```
Failed to update the yarn.lock, please update manually before merging.
```
</details>

##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------
critical severity  |  **776/1000**  <br/> **Why?** Recently disclosed, Has a fix available, CVSS 9.8  | Improper Input Validation <br/>[SNYK-JS-SOCKETIOPARSER-3091012](https://snyk.io/vuln/SNYK-JS-SOCKETIOPARSER-3091012) |  No  | No Known Exploit 

(*) Note that the real score may have changed since the PR was raised.

Check the changes in this PR to ensure they won't cause issues with your project.

------------

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  
🧐 [View latest project report](https://app.snyk.io/org/arduinobot/project/4cfa0cb2-123a-4f88-bde8-659b45f1062c?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/arduinobot/project/4cfa0cb2-123a-4f88-bde8-659b45f1062c?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"695e41e7-cf27-4f27-9ec9-1c176c6dd324","prPublicId":"695e41e7-cf27-4f27-9ec9-1c176c6dd324","dependencies":[{"name":"@theia/core","from":"1.25.0","to":"1.31.1"},{"name":"@theia/keymaps","from":"1.25.0","to":"1.31.1"},{"name":"@theia/terminal","from":"1.25.0","to":"1.31.1"}],"packageManager":"yarn","projectPublicId":"4cfa0cb2-123a-4f88-bde8-659b45f1062c","projectUrl":"https://app.snyk.io/org/arduinobot/project/4cfa0cb2-123a-4f88-bde8-659b45f1062c?utm_source=github&utm_medium=referral&page=fix-pr","type":"auto","patch":[],"vulns":["SNYK-JS-SOCKETIOPARSER-3091012"],"upgrade":["SNYK-JS-SOCKETIOPARSER-3091012"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["pr-warning-shown","priorityScore"],"priorityScoreList":[776]})

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Improper Input Validation](https://learn.snyk.io/lessons/sql-injection/javascript//?loc&#x3D;fix-pr)

### CODE REVIEW

**Comment by kittaakos** [comment]
Unfortunately, it's a useless auto-bump. 😕

**Review by kittaakos — COMMENTED** [review]
IDE2 cannot do much with the PR.

Problems:
* The bot needs to sign the CLI. This is probably easy.
* The auto PR is shiny and tip-top but does not solve anything:
    * First off, it creates a false public warning,
    * Theia dependencies must be updated together, so bumping one or two of them will break the dependency injection as the same module will be under the `node_modules` multiple times (this is how npm hoisting works),
    * The auto PR does not re-generate the `yarn.lock` file, so essentially, after a merge + pull cycle, all the devs will have an outgoing git change, and the new, partial Theia update will be indeterministic as the version is not pinned.

In summary, the way it works now does not solve anything but creates a false public warning on GH.


Closing as invalid. Feel free to reopen and take care of this PR if you disagree and want to handle the version update.
